doc/architecture/adr-0003-testing-philosophy.md

```diff
@@ -0,0 +1,69 @@
+0003 - Testing Philosophy
+=========================
+
+Context
+-------
+
+Different projects have followed different testing approaches on a
```

Comment:
I think we need to make a decision on hard line breaks for documentation because we haven't used them in the past. I prefer them though so I'd be happy to switch.